nodebuilder/p2p/p2p.go

```diff
@@ -205,6 +228,32 @@ func (m *module) PubSubTopics(_ context.Context) ([]string, error) {
 	return m.ps.GetTopics(), nil
 }
 
+func (m *module) Ping(ctx context.Context, peer peer.ID) (time.Duration, error) {
+	res := <-ping.Ping(ctx, m.host, peer) // context is handled for us
```

Comment:
Wow, what the API.

Reply:
And?

Reply:
Yes

Reply:
simple, right? :)

Reply:
Meh, very uncommon. Unintuitive.

Reply:
What particularly about this API you dont like?

Reply:
Basically where you run the goroutine. Outside or inside the call

Reply:
I mean its quite usual to return result channel. It does more for the user, but in return gives less control to user.